test-utils/src/main/java/com/palantir/conjure/java/api/testing/AbstractServiceExceptionAssert.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.api.testing;
+
+import com.palantir.conjure.java.api.errors.ErrorType;
+import com.palantir.logsafe.Arg;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.assertj.core.api.AbstractThrowableAssert;
+import org.assertj.core.util.Throwables;
+
+abstract class AbstractServiceExceptionAssert<T extends AbstractThrowableAssert<T, U>, U extends Throwable>
+        extends AbstractThrowableAssert<T, U> {
+
+    protected AbstractServiceExceptionAssert(U throwable, Class<T> selfType) {
+        super(throwable, selfType);
+    }
+
+    void hasCode(ErrorType.Code code, ErrorType.Code actualCode) {
+        isNotNull();
+        failIfNotEqual("Expected ErrorType.Code to be %s, but found %s", code, actualCode);
+    }
```

Comment:
Instead of defining methods that take the actual value and redefining methods in the implementation, could you just do this instead?
```suggestion
abstract class AbstractServiceExceptionAssert<T extends AbstractThrowableAssert<T, U>, U extends Throwable>
        extends AbstractThrowableAssert<T, U> {

    protected AbstractServiceExceptionAssert(U throwable, Class<T> selfType) {
        super(throwable, selfType);
    }
    
    protected abstract ErrorType actualErrorType();

    public final void hasCode(ErrorType.Code code) {
        isNotNull();
        failIfNotEqual("Expected ErrorType.Code to be %s, but found %s", code, actualErrorType().code();
    }
```
?

I realize that `EndpointServiceException` and `ServiceException` unfortunately don't implement any shared interface (which they probably should to be honest?), but we should still be able to share most of the implementation and thereby also guarantee that they have the exact same methods, without having to redefine them in the concrete implementation either